lib/puppet/ssl/host.rb

```diff
@@ -146,6 +146,20 @@ def certificate
     @certificate
   end
 
+  # PUP-8736 Device certificates are stored in vardir instead of confdir.
+
+  def is_device_certificate?
+    Puppet[:ssldir].start_with?(Puppet[:vardir])
+  end
```

Comment:
Let's make this explicit by adding an optional argument to:

```ruby
def initialize(name, device = false)
  ...
```

And a corresponding reader:

```ruby
attr_reader :device
```

Then if the `ssl` command is in `device` mode, then call `Puppet::SSL::Host.new(options[:certname], true)`.